src/nfa/delta.cc

```diff
@@ -374,6 +375,24 @@ bool Delta::operator==(const Delta& other) const {
     return other_transitions_it == other_transitions_end;
 }
 
+//Returns an iterator to the smallest epsilon, or end() if there is no epsilon
+//Searches from the end of the vector of SymbolPosts, since epsilons are at the end and they are typically few, mostly 1.
+StatePost::const_iterator StatePost::first_epsilon_it(Symbol first_epsilon) const {
+    auto it = end();
+    while (it != begin()) {
+        --it;
+        if (it->symbol < first_epsilon) { //is it a normal symbol already?
+            ++it; // go step back, to the smallest epsilon or end(), and return it
+            return it;
```

Comment:
Refactor to omit having to modify the local variable.
```suggestion
            return it + 1; // Return one step back: the smallest epsilon or end().
```